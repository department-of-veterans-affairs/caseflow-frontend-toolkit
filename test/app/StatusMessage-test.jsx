/* eslint-disable max-statements */

import React from 'react';
import { render } from '@testing-library/react';
// import { expect } from 'chai';
// import { shallow } from 'enzyme';
// import toJson from 'enzyme-to-json';
import StatusMessage from '../../components/StatusMessage';

describe('StatusMessage', () => {
  it('no props', () => {
    const {asFragment} = render(<StatusMessage />);  
    expect(asFragment()).toMatchSnapshot();
  });

  it('wrapInAppSegment', () => {
    const {asFragment} = render(<StatusMessage wrapInAppSegment={false}/>);  
    expect(asFragment()).toMatchSnapshot();
  });
  it('no props with child', () => {
    const {asFragment} = render(
      <StatusMessage>
        <p>Magnificent desolation</p>
      </StatusMessage>);
    expect(asFragment()).toMatchSnapshot();
  });
  it('checklist only', () => {
    const {asFragment} = render(<StatusMessage checklist={['apple', 'banana', 'cherry']} />);
    expect(asFragment()).toMatchSnapshot();
  });  
  it('checklist only with child', () => {
    const {asFragment} = render(
      <StatusMessage checklist={['apple', 'banana', 'cherry']}>
        <p>Magnificent desolation</p>
      </StatusMessage>);
    expect(asFragment()).toMatchSnapshot();
  });
  it('checklist, leadMessageList only', () => {
    const {asFragment} = render(
      <StatusMessage
        checklist={['apple', 'banana', 'cherry']}
        leadMessageList={['argon', 'boron', 'carbon']}
      />);
    expect(asFragment()).toMatchSnapshot();
  });
  it.only('checklist, leadMessageList only with child', () => {
    const {asFragment} = render(
      <StatusMessage checklist={['apple', 'banana', 'cherry']}
      leadMessageList={['argon', 'boron', 'carbon']}>
        <p>Magnificent desolation</p>
      </StatusMessage>);
    expect(asFragment()).toMatchSnapshot();
  });
  //   expect(toJson(shallow(<StatusMessage
  //     checklist={['apple', 'banana', 'cherry']}
  //     leadMessageList={['argon', 'boron', 'carbon']} >
  //     <p>Magnificent desolation</p>
  //   </StatusMessage>))).to.matchSnapshot()
  // );
  it('checklist, messageText only', () =>
    expect(toJson(shallow(<StatusMessage
      checklist={['apple', 'banana', 'cherry']}
      messageText="dummy message text"
    />))).to.matchSnapshot()
  );
  it('checklist, messageText only with child', () =>
    expect(toJson(shallow(<StatusMessage
      checklist={['apple', 'banana', 'cherry']}
      messageText="dummy message text" >
      <p>Magnificent desolation</p>
    </StatusMessage>))).to.matchSnapshot()
  );
  it('checklist, title only', () =>
    expect(toJson(shallow(<StatusMessage
      checklist={['apple', 'banana', 'cherry']}
      title="dummy message text"
    />))).to.matchSnapshot()
  );
  it('checklist, title only with child', () =>
    expect(toJson(shallow(<StatusMessage
      checklist={['apple', 'banana', 'cherry']}
      title="dummy message text" >
      <p>Magnificent desolation</p>
    </StatusMessage>))).to.matchSnapshot()
  );
  it('checklist, leadMessageList, messageText only', () =>
    expect(toJson(shallow(<StatusMessage
      checklist={['apple', 'banana', 'cherry']}
      leadMessageList={['argon', 'boron', 'carbon']}
      messageText="dummy message text"
    />))).to.matchSnapshot()
  );
  it('checklist, leadMessageList, messageText only with child', () =>
    expect(toJson(shallow(<StatusMessage
      checklist={['apple', 'banana', 'cherry']}
      leadMessageList={['argon', 'boron', 'carbon']}
      messageText="dummy message text" >
      <p>Magnificent desolation</p>
    </StatusMessage>))).to.matchSnapshot()
  );
  it('checklist, leadMessageList, messageText, title', () =>
    expect(toJson(shallow(<StatusMessage
      checklist={['apple', 'banana', 'cherry']}
      leadMessageList={['argon', 'boron', 'carbon']}
      messageText="dummy message text"
      title="dummy message text"
    />))).to.matchSnapshot()
  );
  it('checklist, leadMessageList, messageText, title with child', () =>
    expect(toJson(shallow(<StatusMessage
      checklist={['apple', 'banana', 'cherry']}
      leadMessageList={['argon', 'boron', 'carbon']}
      messageText="dummy message text"
      title="dummy message text" >
      <p>Magnificent desolation</p>
    </StatusMessage>))).to.matchSnapshot()
  );
});
