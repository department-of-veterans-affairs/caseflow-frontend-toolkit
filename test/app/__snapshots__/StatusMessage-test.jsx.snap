// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusMessage checklist only 1`] = `
<DocumentFragment>
  <section
    class="cf-app-segment cf-app-msg-screen cf-app-segment--alt"
  >
    <div>
      <h1
        class="cf-msg-screen-heading"
      />
      <p
        class="cf-msg-screen-text"
      />
    </div>
  </section>
</DocumentFragment>
`;

exports[`StatusMessage checklist only with child 1`] = `
<DocumentFragment>
  <section
    class="cf-app-segment cf-app-msg-screen cf-app-segment--alt"
  >
    <div>
      <h1
        class="cf-msg-screen-heading"
      />
      <h2
        class="cf-msg-screen-deck"
      >
        <p>
          Magnificent desolation
        </p>
      </h2>
      <p
        class="cf-msg-screen-text"
      />
    </div>
  </section>
</DocumentFragment>
`;

exports[`StatusMessage checklist, leadMessageList only 1`] = `
<DocumentFragment>
  <section
    class="cf-app-segment cf-app-msg-screen cf-app-segment--alt"
  >
    <div>
      <h1
        class="cf-msg-screen-heading"
      />
      <h2
        class="cf-msg-screen-deck"
      >
        argon
      </h2>
      <h2
        class="cf-msg-screen-deck"
      >
        boron
      </h2>
      <h2
        class="cf-msg-screen-deck"
      >
        carbon
      </h2>
      <p
        class="cf-msg-screen-text"
      />
    </div>
  </section>
</DocumentFragment>
`;

exports[`StatusMessage checklist, leadMessageList only with child 1`] = `
<DocumentFragment>
  <section
    class="cf-app-segment cf-app-msg-screen cf-app-segment--alt"
  >
    <div>
      <h1
        class="cf-msg-screen-heading"
      />
      <h2
        class="cf-msg-screen-deck"
      >
        <p>
          Magnificent desolation
        </p>
      </h2>
      <p
        class="cf-msg-screen-text"
      />
    </div>
  </section>
</DocumentFragment>
`;

exports[`StatusMessage checklist, leadMessageList, messageText only 1`] = `
<DocumentFragment>
  <section
    class="cf-app-segment cf-app-msg-screen cf-app-segment--alt"
  >
    <div>
      <h1
        class="cf-msg-screen-heading"
      />
      <h2
        class="cf-msg-screen-deck"
      >
        argon
      </h2>
      <h2
        class="cf-msg-screen-deck"
      >
        boron
      </h2>
      <h2
        class="cf-msg-screen-deck"
      >
        carbon
      </h2>
      <p
        class="cf-msg-screen-text"
      >
        dummy message text
      </p>
    </div>
  </section>
</DocumentFragment>
`;

exports[`StatusMessage checklist, leadMessageList, messageText only with child 1`] = `
<DocumentFragment>
  <section
    class="cf-app-segment cf-app-msg-screen cf-app-segment--alt"
  >
    <div>
      <h1
        class="cf-msg-screen-heading"
      />
      <h2
        class="cf-msg-screen-deck"
      >
        <p>
          Magnificent desolation
        </p>
      </h2>
      <p
        class="cf-msg-screen-text"
      >
        dummy message text
      </p>
    </div>
  </section>
</DocumentFragment>
`;

exports[`StatusMessage checklist, leadMessageList, messageText, title 1`] = `
<DocumentFragment>
  <section
    class="cf-app-segment cf-app-msg-screen cf-app-segment--alt"
  >
    <div>
      <h1
        class="cf-msg-screen-heading"
      >
        dummy message text
      </h1>
      <h2
        class="cf-msg-screen-deck"
      >
        argon
      </h2>
      <h2
        class="cf-msg-screen-deck"
      >
        boron
      </h2>
      <h2
        class="cf-msg-screen-deck"
      >
        carbon
      </h2>
      <p
        class="cf-msg-screen-text"
      >
        dummy message text
      </p>
    </div>
  </section>
</DocumentFragment>
`;

exports[`StatusMessage checklist, leadMessageList, messageText, title with child 1`] = `
<DocumentFragment>
  <section
    class="cf-app-segment cf-app-msg-screen cf-app-segment--alt"
  >
    <div>
      <h1
        class="cf-msg-screen-heading"
      >
        dummy message text
      </h1>
      <h2
        class="cf-msg-screen-deck"
      >
        <p>
          Magnificent desolation
        </p>
      </h2>
      <p
        class="cf-msg-screen-text"
      >
        dummy message text
      </p>
    </div>
  </section>
</DocumentFragment>
`;

exports[`StatusMessage checklist, messageText only 1`] = `
<DocumentFragment>
  <section
    class="cf-app-segment cf-app-msg-screen cf-app-segment--alt"
  >
    <div>
      <h1
        class="cf-msg-screen-heading"
      />
      <p
        class="cf-msg-screen-text"
      >
        dummy message text
      </p>
    </div>
  </section>
</DocumentFragment>
`;

exports[`StatusMessage checklist, messageText only with child 1`] = `
<DocumentFragment>
  <section
    class="cf-app-segment cf-app-msg-screen cf-app-segment--alt"
  >
    <div>
      <h1
        class="cf-msg-screen-heading"
      />
      <h2
        class="cf-msg-screen-deck"
      >
        <p>
          Magnificent desolation
        </p>
      </h2>
      <p
        class="cf-msg-screen-text"
      >
        dummy message text
      </p>
    </div>
  </section>
</DocumentFragment>
`;

exports[`StatusMessage checklist, title only 1`] = `
<DocumentFragment>
  <section
    class="cf-app-segment cf-app-msg-screen cf-app-segment--alt"
  >
    <div>
      <h1
        class="cf-msg-screen-heading"
      >
        dummy message text
      </h1>
      <p
        class="cf-msg-screen-text"
      />
    </div>
  </section>
</DocumentFragment>
`;

exports[`StatusMessage checklist, title only with child 1`] = `
<DocumentFragment>
  <section
    class="cf-app-segment cf-app-msg-screen cf-app-segment--alt"
  >
    <div>
      <h1
        class="cf-msg-screen-heading"
      >
        dummy message text
      </h1>
      <h2
        class="cf-msg-screen-deck"
      >
        <p>
          Magnificent desolation
        </p>
      </h2>
      <p
        class="cf-msg-screen-text"
      />
    </div>
  </section>
</DocumentFragment>
`;

exports[`StatusMessage no props 1`] = `
<DocumentFragment>
  <section
    class="cf-app-segment cf-app-msg-screen cf-app-segment--alt"
  >
    <div>
      <h1
        class="cf-msg-screen-heading"
      />
      <p
        class="cf-msg-screen-text"
      />
    </div>
  </section>
</DocumentFragment>
`;

exports[`StatusMessage no props with child 1`] = `
<DocumentFragment>
  <section
    class="cf-app-segment cf-app-msg-screen cf-app-segment--alt"
  >
    <div>
      <h1
        class="cf-msg-screen-heading"
      />
      <h2
        class="cf-msg-screen-deck"
      >
        <p>
          Magnificent desolation
        </p>
      </h2>
      <p
        class="cf-msg-screen-text"
      />
    </div>
  </section>
</DocumentFragment>
`;

exports[`StatusMessage wrapInAppSegment 1`] = `
<DocumentFragment>
  <div
    class="cf-app-msg-screen"
  >
    <div>
      <h1
        class="cf-msg-screen-heading"
      />
      <p
        class="cf-msg-screen-text"
      />
    </div>
  </div>
</DocumentFragment>
`;
